### ISSUE TITLE
broken with latest insiders?

### ISSUE THREAD

**Issue by chrisdias** [hoovercj/vscode-power-mode#20]
with latest insiders, the explosions don't follow the cursor, they always appear at the beginning of the line.  i know some work was done to optimize decorators in VS Code last month, maybe something broke?

**Comment by tonyxwz** [comment]
I have this issue too

**Comment by BillYang2016** [comment]
me too

**Comment by SinarPandora** [comment]
me too

**Comment by jshamble** [comment]
me too

**Comment by hoovercj** [comment]
I have no idea why I'm not receiving emails for this repo, but I get them for my other repos. I'm so sorry that this has made it out of insiders and into the stable release (or so I assume from the reviews I'm getting). Unfortunately I'm traveling without a laptop for the holidays but I'll try to have this fixed early next year.

Thanks for your patience!

**Comment by hoovercj** [comment]
Update: the settings issue is resolved and I should be notified for all future activity here and in New issues.

**Comment by Alfiory** [comment]
Hello,
I'm sorry to reopen this issue but I've the same problem but I don't think I've an insider version, i'm in 1.19.1.
I don't know if it's important but I'm on linux (ArchLinux with xfce4).

So can someone help me ? 

Thanks in advance ! (I'm french, so sorry if I don't speak very well ^^)

**Comment by hoovercj** [comment]
@alfiory the "real" issue has not yet been fixed, I haven't come back from holidays yet.

The only "issue" I have fixed was with GitHub notification settings. Apologies for being unclear and happy new year!

**Comment by Alfiory** [comment]
Okay sorry I didn't understand sorry, thanks so.

Thanks ! Happy New Year to you too ;)

**Comment by kirkone** [comment]
Hi,

did some debugging on this an have an idea what has changed.
My assumption is that the way how VSCode handles `::before` and `::after` has changed in the .19 update.
If I use the `after` property in `DecorationRenderOptions` they add a new span after the span with the current sign:
```
<span>
    <span class="mtk1">sdfgsd&nbsp;gsdfg&nbsp;sdffgs&nbsp;sdfgsdfaasdf</span>
    <span class="mtk1 ced-TextEditorDecorationType12-1 ced-TextEditorDecorationType13-1">x</span>
    <span class="ced-TextEditorDecorationType12-4 ced-TextEditorDecorationType13-4"></span>
</span>
```
the `x` is my last typed letter and the `span` after that has the `::after` element:
![image](https://user-images.githubusercontent.com/15778270/34894086-461beb28-f7e0-11e7-8c3a-e09fcbb9fdab.png)
This `span` does not have set `position: relative`. That is why the `::after` element is positioned at the beginning of the line because this is the next element with relative positioning.

At the moment I have no clue how to fix this but I thought I will share my progress so far.

so long,
KirK

**Comment by hoovercj** [comment]
@kirkone provided a patch which fixes this and I've just published it in version 2.1.0.
